### DISCUSSION TITLE
The size of ext plugin is big

### DISCUSSION THREAD

**Discussion by hqnghi88** [gama-platform/gama#78] [General]
If we take a look in the list of plugins artifact, easily to see that the msi.gama.ext is the biggest. It is fine if we download whole GAMA package with high speed internet connection. However in some cases, it is hard to make it work well with low-speed network, or in specific configuration, such as the update mechanism of eclipse (still not work well as sometime it fail to download big artifact), compiling/installing/deploying somewhere somehow with the limitation of files (10mb or 50mb only)...
As we know have increase the modularity of gama, and someday with the auto-installation of required packages, i wonder if we can split that big ext into smaller pieces. I tried to split by 2, mostly there is geotools as the big part and the rest. It seem to have no technical problem.
What do you think?

**Comment by AlexisDrogoul** [comment]
No problem on my side. Splitting this huge plugin into smaller ones should not be an issue.

**Comment by hqnghi88** [comment]
I have push to the branch splitting_ext https://github.com/gama-platform/gama/tree/splitting_ext . The ext is splitted to 3 : msi.gama.ext, msi.gama.ext1 and msi.gama.ext2 , with the size from 60+MB to 30, 15 and 15 MB. It obviously make the updating from eclipse worlk better when downloading smaller packages.
But the name is so ugly (ext1, ext2) i cant found the informative name for that. Any idea?

**Comment by AlexisDrogoul** [comment]
Are you still wanting to push this change? In the meantime, I had split the "dependencies" feature into two features ("dependencies" and "dependencies.ui"), so that headless do not have to depend on UI components. I don't know if it provides any help for you ?

**Comment by hqnghi88** [comment]
Yes the splitting increase the update success rate so i still want to have this change (should rework more on choosing which jar separated in which ext , to give a meaningful names instead of ext1 ext2)
But the splitting dependencies to .ui is not related , i think, because it is the higher level than single jar (which is the objective of this issue: split the big jar msi.gama.ext)

**Reply by lesquoyb** [reply]
I know it has been a long time and we even changed the architecture a bit since then, but it is still an issue @hqnghi88 ?

**Reply by hqnghi88** [reply]
Yes, it's still the same, the new renaming gama.dependencies size AFAIK are big.

**Reply by RoiArthurB** [reply]
I guess it is as we didn't remove any jar from that plugin. However, since we squashed the history, cloning the full repo should be somehow faster.

**Reply by hqnghi88** [reply]
It raised the % of corruption when install/update from Help menu, not in clone nor the build process.

**Comment by lesquoyb** [comment]
Closing the discussion as I turned it into an issue
